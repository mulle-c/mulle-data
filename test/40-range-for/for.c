#include <mulle-data/mulle-data.h>

#include <stdio.h>


static void   mulle_range_print( struct mulle_range a)
{
   if( ! mulle_range_is_valid( a))
      printf( "INVALID");
   else
      if( ! a.length)
         printf( "NULL");
      else
         printf( "%tu-%tu (%tu)", a.location, a.location + a.length - 1, a.length);
}


static void  test( struct mulle_range a)
{
   char   *sep = " ";

   mulle_range_print( a);
   printf( ": ");

   mulle_range_for( i, a)
   {
      printf( "%s%td", sep, i);
      sep =", ";
   }
   printf( "\n");
}


int   main( int argc, char *argv[])
{
   test( mulle_range_make( 0, 0));
   test( mulle_range_make( 0, 1));
   test( mulle_range_make( 2, 4));
   test( mulle_range_make_invalid());

   return( 0);
}


/*
 * extension : mulle-sde/c-test-library-demo
 * directory : demo/all
 * template  : .../hello.c
 * Suppress this comment with `export MULLE_SDE_GENERATE_FILE_COMMENTS=NO`
 */
