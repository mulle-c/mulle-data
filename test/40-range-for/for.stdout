NULL: 
0-0 (1):  0
2-5 (4):  2, 3, 4, 5
INVALID: 
